File: tareas/03/11_martinez_ostoa/tarea06_intent_vs_activity/app/src/main/java/com/example/nestorivanmo/tarea06_intent_vs_activity/MainActivity.java
package com.example.nestorivanmo.tarea06_intent_vs_activity;

import android.content.Intent;
import android.net.Uri;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void muestraActividad(View view){
        Intent intent = new Intent(MainActivity.this, Main2Activity.class);
        startActivity(intent);
    }

    public void muestraNavegador(View view){
        Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("http://www.google.com"));
        startActivity(browserIntent);
    }
}

File: ejemplos/Sensores/app/src/main/java/com/example/angel5215/sensores/MainActivity.kt
package com.example.angel5215.sensores

import android.content.Context
import android.content.Intent
import android.hardware.Sensor
import android.hardware.SensorManager
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        vibrador.setOnClickListener(this)
        acelerometro.setOnClickListener(this)
        giroscopio.setOnClickListener(this)
        proximidad.setOnClickListener(this)

        //  Conocer qué sensores tiene disponibles nuestro teléfono
        val sensorManager = getSystemService(Context.SENSOR_SERVICE) as? SensorManager
        val listaSensores = sensorManager?.getSensorList(Sensor.TYPE_ALL)

        listaSensores?.forEach {
            sensoresTextView.append(it.name + "\n")
        }
    }

    override fun onClick(v: View?) {
        when (v?.id){
            R.id.vibrador -> {
                val intent = Intent(this, VibradorActivity::class.java)
                startActivity(intent)
            }
            R.id.acelerometro -> {
                val intent = Intent(this, AcelerometroActivity::class.java)
                startActivity(intent)
            }
            R.id.giroscopio -> {
                val intent = Intent(this, GiroscopioActivity::class.java)
                startActivity(intent)
            }
            R.id.proximidad -> {
                val intent = Intent(this, ProximidadActivity::class.java)
                startActivity(intent)
            }
        }
    }
}
